### DISCUSSION TITLE
Validation PositiveOrZero is incorrect

### DISCUSSION THREAD

**Discussion by fd6130** [symfony/symfony#43438] [Q&A] [answered]
```php

/**
* @Assert\PositiveOrZero
*/
public string $decimal;
```

When i assign string "test" value, it pass the validation. Is it a bug or just because that validation is for "numbers" only? Thanks.

**Comment by michaoj** [comment, marked as answer]
This occurs because your $decimal variable is a string and a non nul string is more than 0. Change the type to int or instance. Or create you own validator to cast the value as int first.
